library/ziti.c

```diff
@@ -1783,3 +1796,93 @@ void ziti_queue_work(ziti_context ztx, ztx_work_f w, void *data) {
 
     uv_async_send(&ztx->w_async);
 }
+
+int ziti_context_init(ziti_context *ztx, const ziti_config *config) {
+    if (config == NULL || config->controller_url == NULL) {
+        ZITI_LOG(ERROR, "config or controller/tls has to be set");
+        return ZITI_INVALID_CONFIG;
+    }
+
+    ziti_context ctx = calloc(1, sizeof(*ctx));
+
+    if (strncmp(config->id.ca, "file://", strlen("file://")) == 0) {
```

Comment:
I think we need to test `config->id.ca` for null before passing it to `strncmp`.